File: C++/Encode_and_Decode_Strings.cpp
class Codec {

public:
    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        /*
        Using my self-defined two digit 'hex' string, where
        'a'-'o' <==> 0 - 15.
        ',' is the seperator of the string
        */
        string ret("");
        for (string &s : strs) {
            for (char &c : s) {
                ret.append(1, c/16+'a');
                ret.append(1, c%16+'a');
            }
            ret.append(1, ',');
        }
        return ret;
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> ret;
        int tmp = 0;
        bool done = false;
        string x("");
        for (char &c : s) {
            if (c == ',') { 
                ret.push_back(x); 
                x = "";
            }
            else {
                tmp = tmp*16 + c - 'a';
                if (done) {
                    x.append(1, (char)tmp);
                    tmp = 0;
                    done = false;
                }
                else done = true;
            }
        }
        return ret;
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: Java/Encode_and_Decode_Strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder sb = new StringBuilder();
        for (String s : strs) {
            for (int i = 0; i < s.length(); ++i) {
                sb.append((char)(s.charAt(i)/16+'a'));
                sb.append((char)(s.charAt(i)%16+'a'));
            }
            sb.append(',');
        }
        return sb.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> ret = new LinkedList<String>();
        StringBuilder sb = new StringBuilder();
        char c = 0;
        boolean end = false;
        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == ',') {
                ret.add(sb.toString());
                sb = new StringBuilder();
            }
            else {
                if (end) {
                    c += s.charAt(i)-'a';
                    sb.append(c);
                    c = 0;
                    end = false;
                }
                else {
                    c += 16*(s.charAt(i)-'a');
                    end = true;
                }
            }
        }
        return ret;
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));
